lib/Page.js

```diff
@@ -255,11 +255,13 @@ class Page extends EventEmitter {
    * @param {Array<Network.CookieParam>} cookies
    */
   async setCookie(...cookies) {
+    const pageURL = this.url();
+    const startsWithHTTP = pageURL.startsWith('http');
     const items = cookies.map(cookie => {
       const item = Object.assign({}, cookie);
-      const pageURL = this.url();
-      if (!item.url && pageURL.startsWith('http'))
-        item.url = this.url();
+      if (!item.url && startsWithHTTP)
+        item.url = pageURL;
+      console.assert(item.url !== 'about:blank', `Blank page can not have cookie "${item.name}"`);
```

Comment:
I think data urls also cannot have cookies.

Reply:
Fixed in commit f1f83c448f73a7ae1e032b838e250d82cd514055